### PR TITLE
Move create/patch device functions to fleet class, and export types

### PR DESCRIPTION
**Pull request by alenjdev**

### CODE REVIEW

**Comment by alenjdev** [comment]
refactor it by moving certain functions to the Fleet class and making them static. Additionally, we can introduce new functions that allow querying device metadata without the need to create instances of the `Device` class. This modification is beneficial in scenarios where only device metadata is required, as the Device class might contain unnecessary functionality, and lack some data.

File: packages/data-sdk/src/Fleet.ts [hunk 1]
```diff
@@ -130,4 +133,8 @@ export class Fleet {
   static queryDevices = queryDevices;
   static queryEvents = queryEvents;
   static queryTelemetry = queryTelemetry;
+  static createDevice = createDevice;
+  static patchDevice = patchDevice;
+  static getDevicesData = getDevicesData;
+  static queryDevicesData = queryDevicesData;
```
Comment (jholmes-formant) [inline packages/data-sdk/src/Fleet.ts:139]: yea, these are interesting… `Fleet` is kinda a grab-bag of "functions" that don't always seems to relate to any _entity_—if you will—of **fleety-ness**. But I think this is "right", given the current architecture.

You _could_ also approach it as `static` on `Device`:

```ts
class Device {
  static createDevice = createDevice;
  static patchDevice = patchDevice;
  static getDevicesData = getDevicesData;
  static queryDevicesData = queryDevicesData;

  /* ...instance methods and whatnot... */
}
```

Then you _could do_:

```ts
Device.createDevice()
```

whether or not thats better than `Fleet.createDevice()` is… _questionable_.

**Review by jholmes-formant — APPROVED** [review]

File: packages/data-sdk/src/devices/Device.ts [hunk 2]
```diff
@@ -565,7 +565,7 @@ export class Device extends BaseDevice {
     return await createShareLink(share, view);
   }
 
-  async createDevice(device: IDevice): Promise<IDevice> {
+  static async createDevice(device: IDevice): Promise<IDevice> {
```
Comment (jholmes-formant) [inline packages/data-sdk/src/devices/Device.ts]: you can definitely still pull these into those little _api_ functions like the others, and just "export" them statically on `Device`. I think there _are_ some good advantages (at least as we work through some of the "responsibilities" of these singletons/classes and refactor them) to think of them that way.

Comment (alenjdev) [reply packages/data-sdk/src/devices/Device.ts]: yeah I saw the comment after I push the changes, but I just reverted to have them as single files per function and just import them in the class

**Pull request merged**